velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp

```diff
@@ -428,6 +428,31 @@ TEST_F(ParquetTableScanTest, aggregatePushdown) {
   assertEqualVectors(rows->childAt(1), valuesVector);
 }
 
+TEST_F(ParquetTableScanTest, aggregatePushdownToSmallPages) {
+  auto expectedRowVector = makeRowVector(
+      {makeFlatVector<int16_t>({2, 4, 1}),
+       makeFlatVector<int64_t>({35, 3, 1})});
+  auto outputType =
+      ROW({"searchengineid", "isrefresh", "searchphrase"},
+          {SMALLINT(), SMALLINT(), VARCHAR()});
+  auto plan =
+      PlanBuilder(pool())
+          .tableScan(
+              outputType,
+              {},
+              "searchphrase <> '' AND searchengineid in (cast(1 as SMALLINT), cast(2 as SMALLINT), cast(4 as SMALLINT))")
+          .singleAggregation({"searchengineid"}, {"sum(isrefresh) as s"})
+          .orderBy({"s DESC"}, false)
+          .planNode();
+  std::vector<std::shared_ptr<connector::ConnectorSplit>> splits;
+  splits.push_back(makeSplit(
+      getExampleFilePath("clickbench-refreshes-pagesize-128.parquet")));
```

Comment:
Why do we need to check in this file? Can we not create it on the fly? What is the rational for the file name?

Reply:
From my end, the bug was first found when I was testing Gluten with ClickBench queries. ClickBench doesn't provide a dataset generator but has a dataset for downloading:

https://datasets.clickhouse.com/hits_compatible/hits.parquet

The data used in this test is reading from the ClickBench dataset but sampled to a smaller size and with the pagesize reduced to 128 bytes maximumly.

Reply:
Would it be possible to generate a dataset with similar characteristic at runtime to avoid checking in the file?